### ISSUE TITLE
Additional Features Request

### ISSUE THREAD

**Issue by unixneo** [godlikemouse/ForceDirectedLayout#2]
Hi Jason,

First of all, your FDG code is awesome.   It builds very nice graphs.    Thanks so much for this code.

I have a few feature requests to discuss with you; I think they are pretty simple.

(1)  Initial coordinate system:

Discussion:   If we are combining graphs from multiple sources and using a FDG utility such as your excellent utility to process each graph on different distributed servers; we need to insure that the coordinate spaces do not collide when we combine the graphs in a visualization engine.   Perhaps you could add a --init-coords switch to provide initial xyz coords so we can have distinct coordinate space between graphs build separately and combined together?  (See 3 below for an even better idea).

(2) Initial conditions and FDG params set with a config file.

Your FDG params are set in layout.cc, for example:

        width_ = 1024;
        height_ = 768;
        layout_iterations_ = 0;
        attraction_multiplier_ = 5.0f;
        repulsion_multiplier_ = 0.75f;
        max_iterations_ = 1000;
        EPSILON_ = 0.000001f;
        temperature_ = width_ / 10.0f;

To provide more flexibility, perhaps it would be good to also have these vars in a config file which is read when initialized using a switch --config-file  or something like that.  These would override the defaults in layout.cc and of course the default in layout.cc. would be used in there is an error reading the config file or the file does not exist, etc.  (Again, I think this item  is also best covered in 3 below.)

(3) Parameters read in from the graph input file (JSON or XML).

Would be also great (maybe instead of 2 above or in addition to 2) to permit attributes in the input JSON or XML files to be used to set graph params.    Maybe a meta XML node or JSON array or something like that which does the same as the config file in 2 above?     After thinking about this out load in the message, I think (3) is preferable to (2) and also to (1).    However, I am not sure if this conforms with graphml or not.   Seems, however, that graphml would support reading in params for the graph.

In summary,  adding the capability to the input XML or JSON file to provide the FDG  params currently in layout.cc and also adding some params for centering and constraining the coordinate space (so multiple graphs created separately will have their own coordinate space / sector)

Finally, I think when I was reading your well written code that you may have included a param for "size" which can be used to set force etc for individual nodes?   Is this correct?   It would be good to have the XML/JSON files have an optional param for each node based on some node attribute (size, criticality, etc)

Thanks for reading.

What do you think?

Yours, Tim

**Comment by godlikemouse** [comment]
Hi Tim,

I think it's a fine idea.  I'll look into adding load file configuration if GraphML supports it.  Otherwise I'll just add it as a command line configuration parameter.  I will probably be a while before I get around to doing it however, I have other things currently taking precedence.  Thanks.

**Comment by unixneo** [comment]
Thanks Jason!

No worries my friend.   I know by your deep skilled resume you must be very busy in your day job!

Looking forward to the updates, when you get a chance, of course!

Thank you again and feel free to ping me anytime!
